### ISSUE TITLE
automatic man page-based hints on completion on -- picks strange sections of docs to show

### ISSUE THREAD

**Issue by bkerin** [akinomyoga/ble.sh#386]
GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu) [Ubuntu 22.04.3 LTS]
ble.sh, version 0.4.0-devel4+fe6b4787 (noarch) [git 2.34.1, GNU Make 4.3, GNU Awk 5.1.0, API: 3.0 (GNU MPFR 4.1.0, GNU MP 6.2.1)]
bash-completion, version 2.11 (hash:b42f5d6a7ad6d4921ec73838ba54a96d6bd30936, 77071 bytes) (noarch)
locale: LANG=en_US.UTF-8
terminal: TERM=xterm-256color wcwidth=14.0-west/15.0-2+ri, vte:6800 (65;6800;1)

This is a nifty feature that I only finally found out about by accident but it seems to pick wrong parts of the option descriptions to show in the example below.  The layout of the option descriptions in the git man pages looks reasonable visually but maybe it's actually hard to parse or something.  I notice it does seem to work right for ls -- 

Also when the terminal is wide I might rather still have just one option per line and see more of the descriptions, than have it go to three columns as it does.  I haven't used it enough yet to know what I'd like for sure on this though.

```
$ git format-patch --
--bare                       : Treat the repository as a bare repository.…
--config-env=<name>=<envvar> : where the sensitive information can be par…
--exec-path[=<path>]         : will print the current setting and then ex…
--git-dir=<path>             : git -C <path>.
--glob-pathspecs             : 1. Disabling globbing on individual pathsp…
--help                       : git help ....
--html-path                  : Print the path, without trailing slash, wh…
--icase-pathspecs            : 1.
--info-path                  : Print the path where the Info files docume…
--list-cmds=group[,group...] : that have git- prefix), list-<category> (s…
--literal-pathspecs          : 1.
--man-path                   : man(1)) for the man pages for this version…
--namespace=<path>           : environment variable.
--noglob-pathspecs           : 1. Enabling globbing on individual pathspe…
--no-optional-locks          : 0.
--no-pager                   : [-P] Do not pipe Git output into a pager.
--no-replace-objects         : for more information.
--paginate                   : [-p] configuration options (see the "Confi…
--super-prefix=<path>        : Currently for internal use only. Set a pre…
--version                    : --version.
--work-tree=<path>           : for a more detailed discussion).
```

**Comment by akinomyoga** [comment]
Thank you for the report.

The issue that strange descriptions are generated was caused by just a simple mistake. Since the format of the man pages depends on the command, we need to rely on some heuristic detection of the man page formats. The format of `git` man page seems similar to that of `ping`, so the extraction code added for `ping` was populated for the git man page. I intended to support the multiline descriptions with this format, but only the last line of the description (in the Nroff format) was extracted. I didn't notice this bug because all the descriptions of the options of `ping` were single-lined. I've fixed it in commit 90a992cc.

For the behavior that the options for `git` (extracted from `man git`) are shown instead of the ones for `git format-patch` (extracted from `man git-format-patch`), it was just not supported. The autogeneration of the descriptions from the man page was just performed with something equivalent to `man <CommandName>`, but the man pages of git subcommands are actually contained in `man git-<subcommand>`. I've added special treatments for the git subcommands in commit 9641c3b8.

**Comment by bkerin** [comment]
This resolves the issue, thanks.

**Comment by bkerin** [comment]
Resolved.

**Comment by akinomyoga** [comment]
Thanks for the report and the confirmation.
